### ISSUE TITLE
Inconsistent committees

### ISSUE THREAD

**Issue by paulhauner** [sigp/lighthouse#402]
## Description

It appears that the same state (by tree hash root) can give two different results for block proposer.

The following code produces the following output.

#### Code

```rust
dbg!(state.canonical_root());
dbg!(self.chain.current_state().canonical_root());

dbg!(state.get_beacon_proposer_index(slot, RelativeEpoch::Current, &self.spec));
dbg!(self.chain.current_state().get_beacon_proposer_index(
        slot,
        RelativeEpoch::Current,
        &self.spec
));
```

#### Output

```
[beacon_node/beacon_chain/src/test_utils.rs:167] state.canonical_root() = 0xb55c5a1fadacd5d92268e2d2e72f492ca7925dee09b8ab1c810cd0bd4d51a485
[beacon_node/beacon_chain/src/test_utils.rs:168] self.chain.current_state().canonical_root() = 0xb55c5a1fadacd5d92268e2d2e72f492ca7925dee09b8ab1c810cd0bd4d51a485
[beacon_node/beacon_chain/src/test_utils.rs:170] state.get_beacon_proposer_index(slot, RelativeEpoch::Current, &self.spec) = Ok(
    0,
)
[beacon_node/beacon_chain/src/test_utils.rs:171] self.chain.current_state().get_beacon_proposer_index(slot,
                                                     RelativeEpoch::Current,
                                                     &self.spec) = Ok(
    12,
)
```

**Comment by paulhauner** [comment]
I'm looking into this now.

**Comment by paulhauner** [comment]
Update: seems to be an error with `generate_seed` where `epoch == current_epoch + 1`.

**Comment by paulhauner** [comment]
Resolved this -- turns out it was related to #407. Solved in #398 by ensuring that the "next" committee cache must be rebuilt after an epoch transition.
